### PR TITLE
Removing rpkg code from RST

### PR DESCRIPTION
**Pull request by egthomas**
This commit removes everything in the rpkg directory except build.txt, which is moved to build/scripts, as well as several of the rpkg and pkg functions in build/script. The README file in the build/script directory is updated to no longer reference the deleted commands. make.code, make.doc, rst.bash, rst.tcsh, and the .gitignore file are also updated to reflect these changes. Finally, the creation of symbolic links to IDL paths is shifted from the now-deleted rpkg.post-setup.sh to the end of make.code.

To summarize, I've stripped out the self-packaging stuff in the rpkg and build/script directory that only Rob knew how to use; all of the necessary remaining scripts and environment variables have been modified accordingly to ensure successful compilation.

### CODE REVIEW

**Comment by egthomas** [comment]
This pull request is going to have another conflict in build.txt once the usr_cleanup pull request is approved so I'm going to hold off on resolving existing conflicts for now.

**Comment by egthomas** [comment]
Now that the other various "cleanup" pull requests (ros_cleanup, usr_cleanup, java_cleanup, etc) have been approved and merged this last one is finally ready to move forward.

I have tested and confirmed compilation of this branch on Debian 8.7.1, Mint 18.1, OpenSUSE 42.2, and Ubuntu 16.04.2 virtual machines.

**Comment by ksterne** [comment]
Same as usual test here; does it compile from a fresh clone?

**Comment by sshepherd** [comment]
I think Evan's comment is that it already compiles on 4 different virtual machines.

**Comment by ksterne** [comment]
No additional testing needed?

**Comment by egthomas** [comment]
External testing of this pull request is both necessary and welcome.

**Comment by ksterne** [comment]
Do you mean I should try processing fit-level data? Or anything else?  Or is simple "will it compile?" good enough?

**Comment by egthomas** [comment]
Ok, after these last two commits (which ensure the idl/ and include/ sub-directories of codebase/ are not ignored by the .gitignore file) this branch is again ready for testing.  Yes, I think compilation is the only thing that really needs to be confirmed here.  I've simply reorganized the directory structure and shifted a few of the build files/scripts around so they all still work together.  The one noticeable difference will be that the log files no longer contain a version number in their filename, as we are no longer tying this repository to a particular version.

**Comment by ksterne** [comment]
Confirming compilation and seeing the file in `~/log/` doesn't have a 3.5 tag to it.  Looks good, merging it now.
